### PR TITLE
Add Fybrik versions strategy

### LINKED ISSUE #1672: Define Fybrik versions names strategy
- Define Fybrik names strategy
- Define Fybrik CRDs name strategy
- Define the relationship between Fybrik release names and CRDs names.

### PR DESCRIPTION
**Pull request by roytman**
This PR: 
- defines Fybrik names strategy
- defines Fybrik CRDs name strategy
- defines the relationship between Fybrik release names and CRDs names.

/closes #1672 

Signed-off-by: Alexey Roytman <roytman@il.ibm.com>

### CODE REVIEW

File: docs/versionNames.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+## Fybrik Version Policy
+Fybrik uses [semantic versioning model](https://semver.org), which is the module adopted by a wide range of projects,
+e.g. Kubernetes, goLang, Helm and many others.
+
+Fybrik versions are of the form "vX.Y.Z" where X is the major version number, Y is the minor version number 
+and Z is the patch release number. 
+- Major Releases - released less frequently, indicate that **backward-incompatible public API changes**. This release 
+carries no guarantee that it will be backward compatible with preceding major versions.
+- Minor Releases - indicate **backward-compatible public API changes**. This release guarantees backward compatibility 
+and stability.
```
Comment (simanadler) [inline docs/versionNames.md:11]: I would suggest indicating that we support only 2 version backward - assuming that this is what has been decided.

File: docs/versionNames.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+- Patch Releases - should not change main functionality, mostly for security or bug fixes, and therefore guarantee 
+backward compatibility.
```
Comment (simanadler) [inline docs/versionNames.md:13]: Same as previous comment

File: docs/versionNames.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+API objects. The editing process may not be straightforward. The migration may require downtime for applications that 
+rely on the feature.
+- Stable 
+  - The version name is vX where X is an integer. 
+  - The stable versions of features appear in released software for many subsequent versions.
+
+##  Relations between CRD changes and Fybrik releases
+Based on the above definitions, the relation between CRD versions and Fybrik versions is unidirectional. Which means, 
+CRD API changes requires at least minor Fybrik release, on the other hand not each Fybrik release requires a new CRD 
+version.
```
Comment (simanadler) [inline docs/versionNames.md:41]: If we do not yet have a strategy for the other things in Fybrik that have to be versioned, I would suggest stating that they are to be defined - i.e. work in progress.

File: docs/versionNames.md (hunk 1 above)
Comment (roytman) [reply docs/versionNames.md:11]: It is defined in a separate (backward compatibility document), which the first draft will be published soon.

File: docs/versionNames.md (hunk 2 above)
Comment (roytman) [reply docs/versionNames.md:13]: If you mean backward compatibility, it is irrelevant for path releases. They should not change Fybrik behavior, so there is no backward compatibility issue. It is the same for the current releases, e.g. there are v0.6.0 - v0.6.8 releases, they are all compatible.

File: docs/versionNames.md (hunk 3 above)
Comment (roytman) [reply docs/versionNames.md:41]: OK

**Comment by roytman** [comment]
part of #1682

**Pull request closed without merging**